backend/apid/graphql/entity.go

```diff
@@ -62,13 +45,15 @@ func (r *entityImpl) LastSeen(p graphql.ResolveParams) (*time.Time, error) {
 
 // Events implements response to request for 'events' field.
 func (r *entityImpl) Events(p schema.EntityEventsFieldResolverParams) (interface{}, error) {
-	entity := p.Source.(*types.Entity)
+	src := p.Source.(*types.Entity)
+	client := r.factory.NewWithContext(p.Context)
 
 	// fetch
-	ctx := types.SetContextFromResource(p.Context, entity)
-	evs, err := r.eventQuerier.Query(ctx, entity.Name, "")
+	evs, err := fetchEvents(client, src.Namespace, func(obj *types.Event) bool {
+		return obj.Entity.Name == src.Name
+	})
 	if err != nil {
-		return 0, err
+		return evs, err
```

Comment:
Not sure why the return changes here. If `err != nil`, one is not supposed to expect anything out of the other returned parameters anyway. I would say it should return `nil, err`, but I have to admit I don't want to try to fix what's not broken...